homeassistant/components/sensibo/__init__.py

```diff
@@ -43,15 +37,18 @@ async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Change entry unique id from api_key to username
     if entry.version == 1:
         api_key = entry.data[CONF_API_KEY]
-        new_unique_id = await async_validate_api(hass, api_key)
-        if new_unique_id in [INVALID_AUTH, CANNOT_CONNECT, NO_DEVICES, NO_USERNAME]:
+
+        try:
+            new_unique_id = await async_validate_api(hass, api_key)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Thats better but we don't want broad exception handlers since they hide valid exceptions.

```suggestion
        except (AuthenticationError, ConnectionError, NoDevicesError, NoUsernameError):
```